### PR TITLE
fix nested folder import issues

### PR DESCRIPTION
**Pull request by srivastavankit**
in case of nested folder import in another service, ensure that we walk up the path to find the actual service.

Currently a service A can import a thrift from another service's nested folder like this. 

```
a.thrift: 

include ../b/nestedFolder/nestedThrift.thrift

service A {
...
}
```

At the time of checking & fetching dependency, ensure that we walk up the nested folder path to find the correct service.

### CODE REVIEW

File: bin/idl.js [hunk 1]
```diff
@@ -534,6 +534,17 @@ function fetch(service, cb) {
         }
     }
 
+    function findService(json, service) {
+        var path = service.split('/');
```
Comment (kriskowal) [inline bin/idl.js]: Consider using the Node.js paths module.

File: bin/idl.js [hunk 2]
```diff
@@ -534,6 +534,17 @@ function fetch(service, cb) {
...
+        for (var i = path.length; i >= 0; i--) {
+            var fetched = json.remotes[path.slice(0, i).join('/')];
```
Comment (kriskowal) [inline bin/idl.js]: Likewise for joining paths. Take care to ensure that paths are absolute.

File: bin/idl.js (hunk 1 above)
Comment (srivastavankit) [inline bin/idl.js]: path is used for directory/file paths. This is the path stored in the meta.json and need not map directly to a folder structure.

File: bin/idl.js (hunk 2 above)
Comment (srivastavankit) [inline bin/idl.js]: `.join` is a native array method, any reason why path.join would be better?

**Comment by kriskowal** [comment]
Fair enough. This change just needs tests.

**Comment by kriskowal** [comment]
Please change the base to `dev`. We publish on `master`.

File: test/idl.js [hunk 3]
```diff
@@ -488,6 +490,132 @@ TestCluster.test('run `idl publish nested folders`', {
...
+                        }
+                    }
+                }
+            }
+        }
+    }
+}, function t(cluster, assert) {
+
+    var now = Date.now();
+    debugger;
```
Comment (kriskowal) [inline test/idl.js]: debugger

File: test/idl.js [hunk 4]
```diff
@@ -488,6 +490,132 @@ TestCluster.test('run `idl publish nested folders`', {
...
+        publishRemote(cluster, 'S', now + 1000, false),
+        updateRemote(cluster, 'S', now + 2000, false, true),
+        publishRemote(cluster, 'S', now + 3000, false),
+        publishRemote(cluster, 'T', now + 4000, false),
+        publishRemote(cluster, 'U', now + 5000, false),
+        fetchRemote(cluster, 'github.com/org/u', now + 6000, true)
+    ], onResults);
+
+    function onResults(err, results) {
+        debugger;
```
Comment (kriskowal) [inline test/idl.js]: debugger

**Comment by kriskowal** [comment]
debugger statements?

**Pull request merged**